pkg/controller/infinispan/infinispan_controller.go

```diff
@@ -336,6 +334,18 @@ func (r *ReconcileInfinispan) Reconcile(request reconcile.Request) (reconcile.Re
 		return reconcile.Result{}, err
 	}
 
+	// Recover Pods with updated init containers in case of fails
+	for _, pod := range podList.Items {
+		if !kube.IsInitContainersEqual(statefulSet.Spec.Template.Spec.InitContainers, pod.Spec.InitContainers) {
```

Comment:
Why do we check for container equality here? If this is to guard against upgrades, shouldn't we just check the conditions?

Reply:
Also, couldn't we just do `!reflect.DeepEqual(statefulSet.Spec.Template.Spec.InitContainers, pod.Spec.InitContainers)` instead of using `IsInitContainersEqual`?

Reply:
> Why do we check for container equality here? If this is to guard against upgrades, shouldn't we just check the conditions?

It's a special hack for recovery Pod in case of StatefulSet controller logic. If customer set incorrect `sha` or `url` and Pod becomes fails, StatefulSet controller never been updated this Pod and we should remove it and than controller will create a correct one from the StatefulSet configuration.

Reply:
> Also, couldn't we just do `!reflect.DeepEqual(statefulSet.Spec.Template.Spec.InitContainers, pod.Spec.InitContainers)` instead of using `IsInitContainersEqual`?

It's a good point, thanks. In this case this probably should work

Reply:
> It's a good point, thanks. In this case this probably should work

Doesn't work as expected as Pod controller embedding additional resources and structures becomes not match. Separate function need to be implemented for validation.